repo/mk/js.test.cypress.mk

```diff
@@ -0,0 +1,76 @@
+# Adds 'cypress/integration/**/*.js' targets to run jest on a specific cypress/integration/**/*.js file.
+#
+# ------------------------------------------------------------------------------
+#
+# Adds a 'test-cypress' internal target to run all SF_CYPRESS_TEST_FILES (defaults to cypress/integration/**/*.js).
+# The 'test-cypress' target is automatically included in the 'test' target via SF_TEST_TARGETS.
+#
+# The arguments to the cypress executable can be changed via CYPRESS_ARGS.
+# Cypress will by default run on the url defined in its cypress.json file.
+# To run it on another url define WEB_BASE_URL before running the target.
+#
+# ------------------------------------------------------------------------------
+
+WAIT_ON = $(call npm-which,WAIT_ON,wait-on)
+
+KILL_PORT = $(call npm-which,KILL_PORT,kill-port)
+
+CYPRESS = $(call npm-which,CYPRESS,cypress)
+
+CYPRESS_BROWSER_TARGET ?= chrome
+
+CYPRESS_ARGS += \
+	--browser $(CYPRESS_BROWSER_TARGET) \
+
+CYPRESS_PORT ?= 4200
+
+CYPRESS_WEB_URL ?= http://localhost
+
+
+SF_CYPRESS_TEST_FILES += \
+	$(shell $(FIND_Q_NOSYM) cypress/integration -type f -name "*.js" -print) \
+
+SF_VENDOR_FILES_IGNORE += \
+
+SF_CLEAN_FILES += \
+	cypress/videos \
+	cypress/screenshots \
+
+SF_TEST_TARGETS += \
+	test-cypress
+
+WEB_BASE_URL ?=
+
+
+# ------------------------------------------------------------------------------
+
+.PHONY: test-cypress
+test-cypress:
```

Comment:
a test target (smth that runs automatically on `make test`) should have a stable behaviour, but now it's dynamic based on the existence of a general-purpose variable.
i think it's fine if the target is "smart" i.e. start a local server if one doesn't exist (could use the same wait-on), but it's a different thing to run tests against a different server